functorch/functorch/_src/fx_minifier.py

```diff
@@ -239,25 +241,34 @@ def delta_debugging(cur_graph: fx.Graph, cur_inps, granularity):
 
     failing_state = ReproState(failing_graph, inps)
 
-    def try_granularity(failing_state, granularity):
+    def try_granularity(failing_state, granularity, use_non_granular=True):
         print(f"Trying granularity {granularity}")
 
+        strategies = []
         num_nodes = len(failing_state.graph.nodes)
         num_outputs = len(get_outputs(failing_state.graph))
         if num_outputs > num_nodes // 2:
-            new_state = remove_outputs(failing_state, max(granularity // 2, 1))
-            if new_state is not None:
-                return new_state
+            strategies += [remove_outputs]
+
+        if use_non_granular:
+            strategies += [eliminate_dead_code, remove_unused_inputs]
 
-        for strategy in [eliminate_dead_code, remove_unused_inputs, remove_suffix, delta_debugging]:
+        strategies += [remove_suffix, delta_debugging]
+
+        for strategy in strategies:
             new_state = strategy(failing_state, granularity)
             if new_state is not None:
                 return new_state
         return None
 
     while True:
         granularity = int(2**(math.floor(math.log2(len(failing_state.graph.nodes)))))
+        new_state = try_granularity(failing_state, granularity, use_non_granular=True)
+        if new_state is not None:
+            failing_state = new_state
+            continue
 
+        granularity //= 2
         has_progress = False
         while granularity >= 1:
             new_state = try_granularity(failing_state, granularity)
```

Comment:
Do we not want to pass in `use_non_granular` here, since those strategies are unaffected ?

Reply:
Oops, you're right. I think I originally set `use_non_granular=False.`